pkg/limits/frontend/service.go

```diff
@@ -77,24 +91,65 @@ func (s *RingIngestLimitsService) forGivenReplicaSet(ctx context.Context, replic
 	return responses, nil
 }
 
-func (s *RingIngestLimitsService) ExceedsLimits(ctx context.Context, r *logproto.ExceedsLimitsRequest) (*logproto.ExceedsLimitsResponse, error) {
-	req := &logproto.GetStreamUsageRequest{
-		Tenant: r.Tenant,
-	}
-	resps, err := s.forAllBackends(ctx, func(_ context.Context, client logproto.IngestLimitsClient) (interface{}, error) {
-		return client.GetStreamUsage(ctx, req)
+func (s *RingIngestLimitsService) ExceedsLimits(ctx context.Context, req *logproto.ExceedsLimitsRequest) (*logproto.ExceedsLimitsResponse, error) {
+	resps, err := s.forAllBackends(ctx, func(_ context.Context, client logproto.IngestLimitsClient) (*logproto.GetStreamUsageResponse, error) {
+		return client.GetStreamUsage(ctx, &logproto.GetStreamUsageRequest{
+			Tenant: req.Tenant,
+		})
 	})
 	if err != nil {
 		return nil, err
 	}
-	var sum uint64
+
+	maxGlobalStreams := s.limits.MaxGlobalStreamsPerUser(req.Tenant)
+
+	var (
+		activeStreamsTotal uint64
+		uniqueStreamHashes = make(map[uint64]bool)
+	)
 	for _, resp := range resps {
-		sum += resp.Response.(*logproto.GetStreamUsageResponse).ActiveStreams
+		if resp.Response == nil {
```

Comment:
Is there a case where a `resp.Response` can be nil when `err` is also non-nil?

Reply:
No I guess it was just muscle memory to protect from nil rather than considering the code branches. This if-statement is obsolete.